sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -26,15 +26,25 @@ public void Execute(GeneratorExecutionContext context)
 
             // attempt to parse user compilation
             var p = new Parser(context);
+
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue(Constants.BuildProperties.EnableSourceGenProp, out var sourceGenSwitch);
+
+            bool.TryParse(sourceGenSwitch, out bool enableSourceGen);
+
+            if (!enableSourceGen)
+            {
+                return;
+            }
+
```

Comment:
Move the `Parser` instance creation(line 28) here. No need to run that constructor if we are going to return.